File: back end/app/models/booking.py
from datetime import datetime

from app.extensions import db

class Booking(db.Model):
    __tablename__ = "bookings"

    booking_id = db.Column(
        db.Integer,
        primary_key=True
    )

    customer_profile_id = db.Column(
        db.Integer,
        db.ForeignKey("user_profiles.user_profile_id"),
        nullable=False
    )

    event_date = db.Column(
        db.Date,
        nullable=False
    )

    status = db.Column(
        db.String(30),
        nullable=False,
        default="PENDING"
    )

    notes = db.Column(
        db.Text,
        nullable=True
    )

    total_price = db.Column(
        db.Numeric(10, 2),
        nullable=False,
        default=0
    )

    created_at = db.Column(
        db.DateTime,
        nullable=False,
        default=datetime.utcnow
    )

    customer = db.relationship(
        "UserProfile",
        back_populates="bookings"
    )

    booking_items = db.relationship(
        "BookingItem",
        back_populates="booking",
        cascade="all, delete-orphan"
    )
    
    
    

File: frontend/src/types/booking.ts
export interface BookingItemPayload {
  service_id: number;
  quantity: number;
  price_at_booking: string;
  notes?: string | null;
}

export interface BookingCreatePayload {
  event_date: string;
  notes?: string | null;
  items: BookingItemPayload[];
}

export interface BookingItem {
  booking_item_id: number;
  service_id: number;
  quantity: number;
  price_at_booking: string;
  notes: string | null;
}

export interface Booking {
  booking_id: number;
  customer_profile_id: number;
  event_date: string;
  status: string;
  notes: string | null;
  total_price: string;
  created_at: string;
  items: BookingItem[];
}